sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -237,21 +253,22 @@ def _create_pipeline(self, credential, **kwargs):
             config.transport = RequestsTransport(**kwargs)
         policies = [
             QueueMessagePolicy(),
+            config.headers_policy,
```

Comment:
Not sure what these changes in the pipeline policy order were all about. We should probably ignore this for now.
I remove the ContentDecode policy below because this was prematurely decoding the XML, however I don't _think_ messing around with the pipeline will be necessary for doing just a `list_blob_names` API.